internal/commands/app/create_inputs.go

```diff
@@ -131,6 +135,65 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 	return fullPath, nil
 }
 
+func (i *createInputs) resolveTemplateID(ui terminal.UI, client realm.Client, templateID string) (string, error) {
+	if templateID == "" && ui.AutoConfirm() {
+		return "", nil
+	}
+
+	templates, err := client.Templates()
+	if err != nil {
+		return "", err
+	}
+
+	// do not disrupt application creation flow if templates are not
+	// available and user is not specifying a template
+	if templateID == "" && len(templates) == 0 {
+		return "", nil
+	}
+
+	if len(templates) == 0 {
+		return "", fmt.Errorf("unable to find template '%s'", templateID)
+	}
+
+	var resolvedTemplateID string
```

Comment:
could reduce a little indenting and likely avoid this `resolvedTemplateID` var all together:

```go
if templateID != "" {
  for _, template := range templates {
    if template.ID == i.Template {
      return template.ID, nil
    }
  }
  return "", fmt.Errorf("template '%s' not found", templateID)
}

options := ...
templateIDs := ...

// ui.AskOne...
```